src/ocl_utils_mod.f90

```diff
@@ -317,36 +320,23 @@ subroutine init_cmd_queues(nqueues, queues, context, device, &
     logical, intent(in), optional :: enable_profiling, out_of_order
     ! Locals
     integer :: i
-    logical :: profiling_property, out_of_order_property
     integer(c_int32_t) :: ierr
-    integer(c_int64_t), parameter :: no_properties = b'00'
-    integer(c_int64_t), parameter :: both_properties = b'11'
+    integer(c_int64_t) :: properties = b'00' ! Start with empty properties
 
-    if (.not.present(enable_profiling)) then
-        profiling_property = .false.
-    else
-        profiling_property = enable_profiling
+    if (present(enable_profiling)) then
+       if (enable_profiling .eqv. .true.) then
```

Comment:
I didn't even know about `.eqv.`! However, in this case, I think it's fine to do `if(enable_profiling)then` and the same for `out_of_order` below. My (new) understanding is that you only need `.eqv.` when comparing variables of boolean type.

Reply:
Ah you are right, I was over-complicating things here.